lib/class-wp-rest-widgets-controller.php

```diff
@@ -28,16 +28,26 @@ class WP_REST_Widgets_Controller extends WP_REST_Controller {
 	 *
 	 * @param WP_Widget[] $widgets Widget objects.
 	 */
-	public function __construct( $widgets, $registered_widgets ) {
+	public function __construct( $widgets ) {
 		$this->namespace = 'wp/v2';
 		$this->rest_base = 'widgets';
 		$this->widgets = $widgets;
-		$this->registered_widgets = $registered_widgets;
+
+		foreach( $this->widgets as $widget ) {
+			$settings = $widget->get_settings();
+			foreach( $settings as $key => $values ) {
+				$this->instances[] = array(
+					'id' => $widget->id_base . '-' . $key,
+					'array_index' => $key,
+					'id_base' => $widget->id_base,
+					'settings' => $values,
+				);
+			}
+		}
```

Comment:
I suggest that this logic be moved to `get_items()` because otherwise the loop and DB query will be done for every single WP load, even if the `widgets` endpoint is not being accessed.